src/django/api/limits.py

```diff
@@ -79,18 +73,28 @@ def check_contributor_api_limit(at_datetime, c):
         .get_or_create(contributor=contributor)
     try:
```

Comment:
I would create context above the try and then update strategy try/except blocks based on the conditionals via Pythonic setter. In except block, it seems that you need just set the BlankDateLimitation strategy.

Reply:
@vladsha-dev My idea was to concentrate changes between "date" managing for Yearly & Monthly subscriptions to refactoring. I didn't plan to touch api logic in scope of this task.

Reply:
I see your point. Anyway, you need to move the logic of choosing the strategy to the check_contributor_api_limit function.
That's why I just recommended you how to move forward to finish work with this PR. My recommendation is based on the documnetation about the strategy pattern. No more.